flink-runtime/src/main/java/org/apache/flink/runtime/io/network/netty/NettyMessage.java

```diff
@@ -64,26 +64,25 @@
 
 	static final String NO_SECURE_COOKIE = "";
 
-	static final Charset DEFAULT_CHARSET = Charset.forName("utf-8");
-
 	abstract ByteBuf write(ByteBufAllocator allocator) throws Exception;
 
 	abstract void readFrom(ByteBuf buffer) throws Exception;
 
 	// ------------------------------------------------------------------------
 
 	private static ByteBuf allocateBuffer(ByteBufAllocator allocator, byte id, String secureCookie) {
-		return allocateBuffer(allocator, id, secureCookie, 0);
+		return allocateBuffer(allocator, id, 0, secureCookie);
 	}
 
-	private static ByteBuf allocateBuffer(ByteBufAllocator allocator, byte id, String secureCookie, int length) {
+	private static ByteBuf allocateBuffer(ByteBufAllocator allocator, byte id, int length, String secureCookie) {
+		final Charset DEFAULT_CHARSET = Charset.forName("utf-8");
 		secureCookie = secureCookie == null ? "": secureCookie;
```

Comment:
Still think we should never pass a null cookie. Then the check wouldn't be necessary.